lib/sax/parser_feedback_simulator.js

```diff
@@ -82,17 +93,41 @@ ParserFeedbackSimulator.prototype._handleStartTagToken = function (token) {
     else if (tn === $.MATH)
         this._enterNamespace(NS.MATHML);
 
-    else if (this.inForeignContent) {
-        if (foreignContent.causesExit(token))
+    if (this.inForeignContent) {
+        if (foreignContent.causesExit(token)) {
             this._leaveCurrentNamespace();
+            return;
+        }
+
+        var currentNs = this.currentNamespace;
+
+        if (currentNs === NS.MATHML)
+            foreignContent.adjustTokenMathMLAttrs(token);
 
-        else if (foreignContent.isMathMLTextIntegrationPoint(tn, this.currentNamespace) ||
-                 foreignContent.isHtmlIntegrationPoint(tn, this.currentNamespace, token.attrs))
+        else if (currentNs === NS.SVG) {
+            foreignContent.adjustTokenSVGTagName(token);
+            foreignContent.adjustTokenSVGAttrs(token);
+        }
+
+        foreignContent.adjustTokenXMLAttrs(token);
+
+        tn = token.tagName;
+
+        if (!token.selfClosing && (
```

Comment:
Maybe extract this condition into boolean var to get rid of this weird formatting and give it some context?